internal/server/server.go

```
@@ -0,0 +1,211 @@
+	}
+	return &Server{
+		config: config,
+		deps:   deps,
+		clock:  realClock{},
+	}, nil
+}
+
+func (s *Server) Run(ctx context.Context) error {
+	for {
+		if err := ctx.Err(); err != nil {
+			return nil
```

golangci-lint (2.12.2)

[error] error is not nil (line 79) but it returns nil

(nilerr)